lib/src/leak_tracking/leak_tracker.dart

```diff
@@ -161,22 +159,35 @@ LeakSummary checkLeaks() {
     return true;
   }());
 
-  return result ?? LeakSummary({});
+  return await (result ?? Future.value(LeakSummary({})));
 }
 
 /// Returns details of the leaks collected since last invocation.
 ///
 /// The same object may be reported as leaked twice: first
 /// as non GCed, and then as GCed late.
-Leaks collectLeaks() {
-  Leaks? result;
+Future<Leaks> collectLeaks() async {
+  Future<Leaks>? result;
 
   assert(() {
-    final tracker = _theObjectTracker();
-    result = tracker.collectLeaks();
+    result = _theObjectTracker().collectLeaks();
+    return true;
+  }());
 
+  return await (result ?? Future.value(Leaks({})));
+}
+
+/// Returns details of the leaks collected since last invocation.
+///
+/// The same object may be reported as leaked twice: first
+/// as non GCed, and then as GCed late.
```

Comment:
Does this dart doc need to be updated? It looks like it was copied from the `collectLeaks` dart doc above